### DISCUSSION TITLE
Add a condition for writing NEXT_LOCALE to cookies

### DISCUSSION THREAD

**Discussion by kliatskiidenis** [amannn/next-intl#1213] [General]
Good afternoon, can you help me with a question? Next js project with app routing, translated by the Next -intl library, a task has appeared, you need to add a condition before writing NEXT_LOCALE to cookies, namely you need to add a check for document.referrer and another custom cookie.
The reason is that NEXT_LOCALE needs to be written into cookies according to conditions, for example: if the link is from Google (document.referrer === 'https://www.google.com/')  then there is no need to write or rewrite NEXT_LOCALE
Please tell me where this can be done? Maybe there is an example solution or a link in the doc on how to do this, but I couldn’t find it

**Comment by factum22** [comment]
Hello community. I need help with a similar task. I will be glad for your advice. Thank you in advance.

**Reply by damian-balas** [reply]
example:
```ts
const handleI18nRouting = (
  options?: Partial<Parameters<typeof createIntlMiddleware>[0]>,
) =>
  createIntlMiddleware({
    ...i18n,
    pathnames: pathnames,
    alternateLinks: false, 
    localeDetection: false,
    ...options,
  });

...

  const hasPreferencesConsent = hasCookieConsent('preferences');

  const response = handleI18nRouting({
    localeDetection: hasPreferencesConsent,
  })(request);
```

`localeDetection` is a option that can be passed to `createIntlMiddleware`. If `localeDetection` is false, the NEXT_LOCALE will not be set.
